### PR TITLE
#168781713 - Users can view travelled Destinations

### PR DESCRIPTION
**Pull request by Cheza-Dzabala**
#### What does this PR do?

This PR enables authenticated users to view the top 5 most travelled locations in the barefoot nomad system. 

#### Description of Task to be completed?

- Allow users and managers to view the top 5 travelled locations in the barefoot nomad system.

#### How should this be manually tested?

1. Clone this repository
2. Run ` sequelize db:migrate `
3. Run ` sequelize db:seed:All `
4. Run ` npm run dev `
5. Log into the application at `api/v1/users/login`
6. Using an  authenticated user send a GET request to `/api/v1/destinations/top-locations`
7. It should display the top 5 locations visited in barefoot nomad. 
8. Incase 4 locations are displayed. Manually change the status of one or more requests to status id `3` and run the endpoint again

#### Any background context you want to provide?

N / A

#### Screenshots (if appropriate)

Screen Shot 2019-11-05 at 08 18 23
Screen Shot 2019-11-05 at 08 18 30
Screen Shot 2019-11-05 at 08 18 35

#### Questions:
N / A

### CODE REVIEW

**Review by sabin18 — CHANGES_REQUESTED** [review]
good job @cheza , here are some changes that you can make on this PR:
-  change message to `Top 5 visited locations`
- There is a middleware which check supplier role use it to check supplier instead of creating new one.

File: src/tests/adminTest.spec.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ describe('Admin Test', () => {
         done();
       });
   });
-  it('Should asign role to user twice', done => {
+  xit('Should asign role to user twice', done => {
```
Comment (nignanthomas) [inline src/tests/adminTest.spec.js]: Why do you `xit` this test?
Kindly put it back to its previous state `it`.

**Review by nignanthomas — CHANGES_REQUESTED** [review]
Great job @Cheza-Dzabala 
I left a comment

File: src/controllers/destinationController.js [hunk 2]
```diff
@@ -27,6 +31,40 @@ export default class destinationController {
       response,
       201
     );
+  }
+
```
Comment (nakiwuge) [inline src/controllers/destinationController.js]: remove this line

File: src/controllers/destinationController.js [hunk 3]
```diff
@@ -27,6 +31,40 @@ export default class destinationController {
...
+  static async viewTopDestinations(req, res) {
+
```
Comment (nakiwuge) [inline src/controllers/destinationController.js]: remove this line

File: src/routes/api/destinations.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Router } from 'express';
+import validateToken from '../../middlewares/auth/validateToken';
+import { verifyPrivileges } from '../../middlewares/allDestinationMiddleware';
+import destinationController from '../../controllers/destinationController';
+
+const router = new Router();
+
+router.use(validateToken, verifyPrivileges);
+
+router.get('/top-locations', (req, res) => destinationController.viewTopDestinations(req, res));
```
Comment (nakiwuge) [inline src/routes/api/destinations.js]: Change `/top-location` to most visited

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@Cheza-Dzabala great job. Kindly check out the comments I have left on this pr

**Review by sabin18 — APPROVED** [review]

**Review by nignanthomas — APPROVED** [review]

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**